modules/analysis-common/src/test/java/org/opensearch/analysis/common/CompoundAnalysisTests.java

```diff
@@ -63,17 +67,25 @@
 import static org.hamcrest.Matchers.instanceOf;
 
 public class CompoundAnalysisTests extends OpenSearchTestCase {
+
+    Settings[] settingsArr;
+
+    @Before
+    public void initialize() throws IOException {
+        this.settingsArr = getSettingsArr();
+    }
```

Comment:
Thanks @hasnain2808 for adding this initialize method. Can we add most of the code inline to prevent developers from accidentally invoking getSettingsArr?

```
@Before
public void initialize() throws IOException {
    final Path home = createTempDir(); 
    copyHyphenationPatternsFile(home);
    this.settingsArr = new Settings[] { getJsonSettings(home), getYamlSettings(home) };
}
```

Reply:
Done

Reply:
Done @jainankitk